File: csharp/log-levels/LogLevels.cs
static class LogLine
{
    public static string Message(string logLine)
    {
        int separatorIndex = logLine.IndexOf(":");
        int messageStartIndex = separatorIndex + 1;
        return logLine[messageStartIndex..].Trim();
    }

    public static string LogLevel(string logLine)
    {
        int levelStartIndex = 1;
        int levelEndIndex = logLine.IndexOf("]");
        return logLine[levelStartIndex..levelEndIndex].ToLowerInvariant();
    }

    public static string Reformat(string logLine)
    {
        return $"{Message(logLine)} ({LogLevel(logLine)})";
    }
}

File: cpp/log-levels/log_levels.cpp
#include <string>

namespace log_line {
	const std::string separator = ": ";

	std::string message(std::string line) {
		size_t separatorIdx = line.find(separator);
		return separatorIdx == std::string::npos
			? ""
			: line.substr(separatorIdx + separator.length());
	}

	std::string log_level(std::string line) {
		size_t separatorIdx = line.find(separator);
		return separatorIdx == std::string::npos
			? ""
			: line.substr(1, separatorIdx - 2);
	}

	std::string reformat(std::string line) {
		return message(line) + " (" + log_level(line) + ")";
	}
} // namespace log_line
